### PR TITLE
Improve coarse-grained parallelism

### PR DESCRIPTION
**Pull request by dzenanz**
This makes better use of available processing power, by overlapping different phases of computation: memory-bound, cache-bound (and if tiles are not in memory, IO).

### CODE REVIEW

**Comment by dzenanz** [comment]
@thewtex Azure pipeline was not run. Probably the same problem as with main ITK repository.

**Comment by dzenanz** [comment]
Module CI needs updating. PR: https://github.com/InsightSoftwareConsortium/ITKContinuousIntegration/pull/30.

**Comment by dzenanz** [comment]
This passes now, after [CI image](https://github.com/InsightSoftwareConsortium/ITKContinuousIntegration/pull/30) has been updated.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 variables:
-  ITKGitTag: v5.0.0
```
Comment (thewtex) [inline azure-pipelines.yml]: I believe we need to support arbitrary Git tags correctly in the Azure configuration. I will break this change out into a separate topic first.

File: include/itkMaxPhaseCorrelationOptimizer.h [hunk 2]
```diff
@@ -88,6 +88,12 @@ class ITK_TEMPLATE_EXPORT MaxPhaseCorrelationOptimizer
   };
   itkGetConstMacro( PeakInterpolationMethod, PeakInterpolationMethod );
   void SetPeakInterpolationMethod( const PeakInterpolationMethod peakInterpolationMethod );
+  friend std::ostream&
+  operator<<( std::ostream& os, const PeakInterpolationMethod& pim )
+  {
+    os << static_cast< typename std::underlying_type< PeakInterpolationMethod >::type >( pim );
+    return os;
+  }
```
Comment (thewtex) [inline include/itkMaxPhaseCorrelationOptimizer.h]: Interesting!

Comment (dzenanz) [reply include/itkMaxPhaseCorrelationOptimizer.h]: I got it from some StackOverflow discussion.

**Pull request merged**